### DISCUSSION TITLE
Pressure Transducer

### DISCUSSION THREAD

**Discussion by Telemachus1977** [tagyoureit/nodejs-poolController#635] [General]
Using this Pressure Transducer from Amazon:

Sender Sensor for Oil Fuel Air Water,1/8"NPT Thread Stainless Steel (100 PSI) https://a.co/d/5rtLmen

And this cheap ADC board from Amazon:

NOYITO 4-Channel 16-Bit ADC with PGA for RPI Raspberry PI 16 Bits I2C ADS1115 Module ADC 4 Channel for Raspberry Pi 3 2 Model B B+ https://a.co/d/2hg9T9O

We were able to accurately get pressure readings on my friend's setup using 3.3V without using a level shifter.  We specifically chose 100psi in case it did need 5V for the logic, so that the pressures in the pool range would be well under 3.3V, but that ended up being unnecessary.

We setup in the ADC menu with a 4.096v power gain.  Maybe another number would be more appropriate here?

![44845](https://user-images.githubusercontent.com/25725743/201764397-4671cac1-a5d7-449e-b961-9774ef5f01dc.jpeg)

Feeding the voltage to a generic pressure transducer:

![44846](https://user-images.githubusercontent.com/25725743/201765088-f94ca692-0f75-40ad-9768-46af2d38ac18.jpeg)

With the pump off and the filter bled (0 PSI), we got .269v from the probe (assuming this is the y intercept?) So we put 0.3v as the inducer offset.  (We expected this to be 0.5V from the limited data available on the Amazon page, but that is based on the expected range of 0.5V=0 pSI and 4.5v=100 PSI, but that is based on 5V input voltage.)

We then checked the PSI on the analog filter guage with the pump at different speeds, and found the slope of the line to be about 40PSI per volt.

For the pressures under 30 that the pool pump runs at, this gave pretty reliable readings.

Feeding that generic device to the filter pressure in njs-PC shows up on the home page.  We tested the flow sensor for REM chem controller by just unplugging the cable from the sensor,  and the warning popped up shortly after.

![44844](https://user-images.githubusercontent.com/25725743/201767064-7972f93b-9cbb-4aca-97d7-a173c4fb8458.jpeg)

Pretty cool!

I have no clue if this was the proper way to set this up but happy with the results so far.

**Comment by rstrouse** [comment]
Now all you have to do is set up the clean/dirty pressure for the filter and it will tell you what percentage of clean the filter is.

**Reply by Telemachus1977** [reply]
There is mention under "high pressure alarm" in REM of an alarm pin.  Is there a way to set a specific gpio pin as an alarm?

**Comment by arrmo** [comment]
OK, this is crazy! Trying to get back to this (after I get my heater going again ... LOL!), but I have the same pressure transducer. Will try to help out here, if I can 😄.

One question, as my wife threw out my wiring diagram, when she was cleaning up for Thanksgiving ... arrgh! Do you happen to have the pinout on the connector (on the transducer side of things)? Make sense?

Thanks!

**Reply by Telemachus1977** [reply]
Red is positive DC+.  Black is DC-.  Whatever the third color wire is is the signal wire.

But how can you do this? Aren't you running the software on a laptop distant from the equipment pad?

**Reply by arrmo** [reply]
> Red is positive DC+. Black is DC-. Whatever the third color wire is is the signal wire.

Yes - agreed. Sorry, my question wasn't very clear 😞. I mean on the actuat connector (end), not the wires. Make sense? Thanks!



> But how can you do this? Aren't you running the software on a laptop distant from the equipment pad?

Yep, I am 😜. But ... I have a CAT5e cable for RS-485 (house builder put that in long ago, easier for them to use the same cabling as they had elsewhere). So I'm stealing some unused lines in that cable.

**Reply by Telemachus1977** [reply]
So how do you plan to connect it to the laptop?  USB ADC?

No idea on the pinout of the plugin.  You'd have to probe it out.

You really should take Tag's offer of the raspberry pi 3.

**Reply by arrmo** [reply]
> So how do you plan to connect it to the laptop? USB ADC?

Yes, correct - USB.

> No idea on the pinout of the plugin. You'd have to probe it out.

NP, will do!

> You really should take Tag's offer of the raspberry pi 3.

And I may 😆. But it doesn't fully fix my issue - or at least, adds others. I don't have network out by the OCP, and also no place to install the RPi - have always had it "remote" (about 30 ft away), as there is no place to put it ... also even WiFi is very poor by the OCP, thanks to brick walls and radiant barrier shielding.

**Reply by Telemachus1977** [reply]
Well like you said you got eight cables to choose from on your cat 5.  Those cables could be coming off of the pi just as easily as the laptop.  

I wonder if USB ADC is supported by the software?

Or heaven forbid to use the cat 5 for wired LAN?

The raspberry Pi stack will take a very small project box.

Or like me, you just stuff it in the low voltage compartment of the Pentair equipment.

**Reply by arrmo** [reply]
> Well like you said you got eight cables to choose from on your cat 5. Those cables could be coming off of the pi just as easily as the laptop.

Sort of 😜. 4 of those wires go to my ICP => only 4 lines left 😞.

> Or heaven forbid to use the cat 5 for wired LAN?

I'd like to do that! If I had all 8 lines available.

> Or like me, you just stuff it in the low voltage compartment of the Pentair equipment.

Agreed, still trying to see if I think I can do that 😄.

**Reply by Telemachus1977** [reply]
Well you only need two pair of cables for a wired ethernet connection...   That CAT5 is perfectly capable of running your ICP and sending internet to your pad.  Just need some CAT5 ends and the appropriate crimpers.

**Reply by tagyoureit** [reply]
Use the existing cable to pull a second line?  Is the conduit big enough?

**Reply by arrmo** [reply]
> Well you only need two pair of cables for a wired ethernet connection... That CAT5 is perfectly capable of running your ICP and sending internet to your pad. Just need some CAT5 ends and the appropriate crimpers.

Ya, I was thinking about that too - don't need high speed 🤣. Let me dig a bit more on that one, I may be able to jury rig something. And I have time, waiting for my transceivers to come in ... LOL.



> Use the existing cable to pull a second line? Is the conduit big enough?

Conduit? Surely you jest ... LMAO. Builder - just put the CAT5 directly in the mortar ... that's their "conduit".

**Reply by Telemachus1977** [reply]
![Screenshot_20221130_135741_Chrome](https://user-images.githubusercontent.com/25725743/204916926-e51ee88b-b082-4ef9-b623-a0f2d3194cb2.jpg)

Obviously the color of the wire doesn't matter as long as the pinout is correct.

https://youtu.be/RxyRBphMO1A

**Reply by arrmo** [reply]
Yep, agreed! 100BASE-TX
https://en.wikipedia.org/wiki/Fast_Ethernet#Copper

**Comment by arrmo** [comment]
And, this - I tend to use 586B, but either way, those pins are pairs (as needed / desired),
![image](https://user-images.githubusercontent.com/8756384/204923144-cd342879-635d-4665-9bf5-80e6df0df70b.png)

**Comment by arrmo** [comment]
OK, this is working awesome - thanks!! Found my cabling issue, now up and running,
![image](https://user-images.githubusercontent.com/8756384/208794938-c32ac27f-cc21-4950-b787-fe55a7bbb749.png)

So cool 😎

**Reply by Telemachus1977** [reply]
Using extra wires from your CAT-5 cable?

**Reply by arrmo** [reply]
Yep! Ran the 3 lines back, connected to the ADC. Really is awesome - thanks!

**Reply by Telemachus1977** [reply]
I agree.  Now how about those darlingtons, lol.

The heater "rule" no longer works on my test array.  I haven't dug into it much.

Edit:. I figured out what happened. I changed the name of the rule and save the interface, but did not disable it and re-enable it.

**Reply by arrmo** [reply]
> I agree. Now how about those darlingtons, lol.

LOL. Ya, if / when I have to rip the main board out (again 😜). Not going to remove it all just for those, as the system is all working now ... LMAO!



> Edit:. I figured out what happened. I changed the name of the rule and save the interface, but did not disable it and re-enable it.

Glad it's working (again)!

**Reply by Telemachus1977** [reply]
Well... Mine is just a Pi zero W hanging from a USB cable off on my bedroom TV.

So technically it's doing absolutely nothing.

I try not to mess with my ACTUAL pool controller if at all possible.

It is much more fun to mess with yours vicariously!

**Reply by arrmo** [reply]
LOL - no worries, I'd say the same too 🤣 

Need to figure out how to PM you - curious about your background 😆 

Have a great holiday break!

**Reply by Telemachus1977** [reply]
My background is pretty disappointing.

I am self taught by YouTube and Google.

I wish I were an electrical engineer.  

Someday I want to get my masters in EE so I can finally start to "understand" how on Earth these things work.

I can't code worth crap but wish I could.

YouTube and Google are slowly teaching me, though.

Someday I want to get my masters in CS so I can finally start to "understand" how on Earth these programs work.

Oh, and I'm not really good at my day job either.

But I already have a terminal degree in that field.

And still rely on YouTube and Google on a regular basis...

😜

**Reply by arrmo** [reply]
Hey, a lot of times - self taught can be better than formal, depending on the individual 👍. To me ... attitude and aptitude go a long ways!

I've got a BSc and MSc in EE ... from quite a ways back (BSc '88, MSc '90). Ya, that part hurts ... LOL.

**Reply by Telemachus1977** [reply]
The thing is, I feel like with enough time I could get a good grasp of electrical engineering.  A lot of it just simply makes "sense."

Coding a piece of software like these gentlemen have done feels about as likely as going to the olympics.

**Reply by arrmo** [reply]
Both very logical, just different logic. As tag says, spinner turns the other direction 🤣

**Comment by arrmo** [comment]
FYI, from Grafana 😄 
![image](https://user-images.githubusercontent.com/8756384/208942574-c8f78642-eb93-4fa0-ac28-0174a06ab051.png)

**Reply by Telemachus1977** [reply]
I've not yet setup grafana.  My buddy tried but didn't make it very far.

**Reply by arrmo** [reply]
Yell if you have any questions - more than happy to help you out! Do you have the data going to influx?

**Reply by Telemachus1977** [reply]
Me... I live in the moment. I try not to look too much into the past or plan too much for the future.  😉

I will see if he wants to join the discussion as he is the one who wants the features.

**Comment by spifftek70** [comment]
I have a Raspberry Pi 4, stacked with Sequent Mega-Bas and 8-Relay board.  I want to add transducers for both filters (pool and spa).
Will this work?  From what I've read, I'll do the analog to digital, plus the level shifting for 3.3v.  Thoughts?

https://www.amazon.com/Bidirectional-Voltage-Level-Shifter-Raspberry-Converter/dp/B0979BSYCW/ref=asc_df_B0979BSYCW/?tag=hyprod-20&linkCode=df0&hvadid=692875362841&hvpos=&hvnetw=g&hvrand=4398813932322713388&hvpone=&hvptwo=&hvqmt=&hvdev=c&hvdvcmdl=&hvlocint=&hvlocphy=9199160&hvtargid=pla-2281435177378&psc=1&mcid=c0702d6a3e8333ada66fc7f6d0965e07&hvocijid=4398813932322713388-B0979BSYCW-&hvexpln=73&gad_source=1

**Reply by tagyoureit** [reply]
Answered here: https://github.com/tagyoureit/nodejs-poolController/discussions/995#discussioncomment-10004164
